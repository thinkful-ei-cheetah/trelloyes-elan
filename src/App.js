import React, { Component } from 'react';
import './App.css';
import STORE from './store';
import List from './composition/List';



class App extends Component {
  render() {
    const listArray = STORE.lists;
    const newListArray = listArray.map((item => <List key={item.id} header={item.header} cards={item.cardIds.map(id => STORE.allCards[id])}/> ))
    return (
        <main className="App">
          <header className="App-header">
            <h1>Hello Trelloyes!</h1>
          </header>
          <div className="App-list">
            {newListArray}
          </div>
        </main>
    )
  }
}

export default App;
