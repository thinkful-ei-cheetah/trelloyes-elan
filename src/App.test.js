import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Card from './composition/Card';
import STORE from './store';
import renderer from 'react-test-renderer';

describe('App component', () => {
  // SMOKE
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  // SNAPSHOT
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<App store={STORE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });  
});





// Card.test.js
describe('Card component', () => {
  // SMOKE
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  // SNAPSHOT
  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<Card title="Test" content="Lorem ipsum" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

