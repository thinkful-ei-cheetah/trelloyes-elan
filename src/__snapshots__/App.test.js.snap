// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Hello Trelloyes!
    </h1>
  </header>
  <div
    className="App-list"
  >
    <div>
      <section
        className="list"
      >
        <header>
          First list
        </header>
        <div
          className="List-cards"
        >
          <div
            className="Card"
          >
            <h3>
              First card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Second card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Fifth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Sixth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Seventh card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Tenth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Twelth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Thirteenth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
        </div>
      </section>
    </div>
    <div>
      <section
        className="list"
      >
        <header>
          Second list
        </header>
        <div
          className="List-cards"
        >
          <div
            className="Card"
          >
            <h3>
              Second card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Third card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Fourth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Sixth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Eighth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Ninth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Eleventh card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
        </div>
      </section>
    </div>
    <div>
      <section
        className="list"
      >
        <header>
          Third list
        </header>
        <div
          className="List-cards"
        >
          <div
            className="Card"
          >
            <h3>
              First card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Second card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Third card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Fourth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Fifth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Sixth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Seventh card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Eighth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Ninth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Tenth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Eleventh card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Twelth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Thirteenth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
        </div>
      </section>
    </div>
    <div>
      <section
        className="list"
      >
        <header>
          Fourth list
        </header>
        <div
          className="List-cards"
        >
          <div
            className="Card"
          >
            <h3>
              Twelth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
          <div
            className="Card"
          >
            <h3>
              Thirteenth card
            </h3>
            <p>
              lorem ipsum
            </p>
          </div>
        </div>
      </section>
    </div>
  </div>
</main>
`;

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    Test
  </h3>
  <p>
    Lorem ipsum
  </p>
</div>
`;

exports[`List component renders the UI as expected 1`] = `
<div>
  <section
    className="list"
  >
    <header>
      test
    </header>
    <div
      className="List-cards"
    >
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
      <div
        className="Card"
      >
        <h3 />
        <p />
      </div>
    </div>
  </section>
</div>
`;
